core/templates/pages/skill-editor-page/services/question-creation-service.spec.ts

```diff
@@ -0,0 +1,145 @@
+    language_code: 'en',
+    version: 1,
+    misconception_count: 3,
+    worked_examples_count: 3,
+    skill_model_created_on: 1593138898626.193,
+    skill_model_last_updated: 1593138898626.193
+  };
+
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.inject(function($injector) {
+    qcs = $injector.get('QuestionCreationService');
+    SkillEditorStateService = $injector.get('SkillEditorStateService');
+    SkillObjectFactory = $injector.get('SkillObjectFactory');
+    SkillDifficultyObjectFactory = $injector.get(
+      'SkillDifficultyObjectFactory');
+    SkillBackendApiService = $injector.get('SkillBackendApiService');
+    EditableQuestionBackendApiService = $injector.get(
+      'EditableQuestionBackendApiService');
+    $uibModal = $injector.get('$uibModal');
+    var misconceptionDict = {
+      feedback: 'feedback',
+      id: 'id1',
+      must_be_addressed: false,
+      name: 'name1',
+      notes: 'notes1'
+    };
+    var rubricDict = {
+      difficulty: 'Easy',
+      explanations: ['Easy']
+    };
+    var conceptCardDict = {
+      explanation: {content_id: 'content',
+        html: 'html_data'},
+      worked_examples: [],
+      recorded_voiceovers: {
+        voiceovers_mapping: {
+          explanation: {},
+          worked_example_1: {},
+          worked_example_2: {}
+        }
+      } };
+    var skillBackendDict = {
+      all_questions_merged: true,
+      description: 'description1',
+      id: 'skillId1',
+      language_code: 'en',
+      misconceptions: [misconceptionDict],
+      next_misconception_id: '2',
+      prerequisite_skill_ids: [],
+      rubrics: [rubricDict],
+      skill_contents: conceptCardDict,
+      superseding_skill_id: 'skillId2',
+      version: 2,
+    };
+
+    spyOn(SkillBackendApiService, 'fetchSkill').and.returnValue({
+      skill: skillBackendDict,
+      topicName: 'topic1',
+      subtopicName: 'subtopic2',
+    });
+    spyOn(SkillBackendApiService, 'fetchMultiSkills').and.returnValue(
+      Promise.resolve([skillBackendDict]));
+    spyOn(SkillEditorStateService, 'getGroupedSkillSummaries').and.returnValue({
+      current: [],
+      others: [skillSummaryDict]
+    });
+
+    $httpBackend = $injector.get('$httpBackend');
+    QuestionObjectFactory = $injector.get('QuestionObjectFactory');
+
+    $rootScope = $injector.get('$rootScope');
+    $location = $injector.get('$location');
+  }));
+
+  it('should create question', function() {
+    var skillDiff = SkillDifficultyObjectFactory.create(
+      'skillId1', 'description', 'Easy');
+    var modalSpy = spyOn($uibModal, 'open').and.returnValue(
+      {result: Promise.resolve([skillDiff])});
+    qcs.createQuestion();
+    expect(modalSpy).toHaveBeenCalled();
```

Comment:
Should you also check if the modal closed with the correct values? Eg. after the modal closes newQuestionSkillIds, newQuestionSkillDifficulties should be set.

Reply:
I have an issue filed for this. will surely improve this there.